hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/DefaultS3ClientFactory.java

```diff
@@ -390,4 +487,68 @@ protected static AmazonS3 configureAmazonS3Client(AmazonS3 s3,
         endpoint, epr, region);
     return new AwsClientBuilder.EndpointConfiguration(endpoint, region);
   }
+
+  /**
+   * Given a endpoint string, create the endpoint URI.
+   *
+   * @param endpoint possibly null endpoint.
+   * @param conf config to build the URI from.
+   * @return an endpoint uri
+   */
+  private static URI getS3Endpoint(String endpoint, final Configuration conf) {
+
+    boolean secureConnections = conf.getBoolean(SECURE_CONNECTIONS, DEFAULT_SECURE_CONNECTIONS);
+
+    String protocol = secureConnections ? "https" : "http";
+
+    if (endpoint == null || endpoint.isEmpty()) {
+      // the default endpoint
+      endpoint = CENTRAL_ENDPOINT;
+    }
+
+    if (!endpoint.contains("://")) {
+      endpoint = String.format("%s://%s", protocol, endpoint);
+    }
+
+    try {
+      return new URI(endpoint);
+    } catch (URISyntaxException e) {
+      throw new IllegalArgumentException(e);
+    }
+  }
+
+  /**
+   * Get the bucket region.
+   *
+   * @param region AWS S3 Region set in the config. This property may not be set, in which case
+   *               ask S3 for the region.
+   * @param credentialsProvider Credentials provider to be used with the default s3 client.
+   * @return region of the bucket.
+   */
+  private Region getS3Region(String region, AWSCredentialsProvider credentialsProvider) {
+
+    if (!StringUtils.isBlank(region)) {
+      return Region.of(region);
+    }
+
+    try {
+      // build a s3 client with region eu-west-2 that can be used to get the region of the bucket.
+      S3Client s3Client = S3Client.builder().region(Region.EU_WEST_2)
```

Comment:
`eu-west-2` / London seems a bit of a random decision.

Let's be intentional by using the [`AWS_GLOBAL`](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/regions/Region.html#AWS_GLOBAL) region value - this will result in `us-east-1`. (Only works for commercial regions. Not sure if we can do anything better for China / US Gov Cloud?)

Reply:
documented in later commit. `"us-east-1"`/`AWS_GLOBAL` can't be used. thanks Ahmar